sk-csharp-azure-functions/config/KernelBuilderExtensions.cs

```diff
@@ -13,11 +13,25 @@ internal static KernelBuilder WithCompletionService(this KernelBuilder kernelBui
         switch (kernelSettings.ServiceType.ToUpperInvariant())
         {
             case ServiceTypes.AzureOpenAI:
```

Comment:
Suggest using when in this scenario:

```suggestion
            case ServiceTypes.AzureOpenAI when kernelSettings.EndpointType == EndpointTypes.TextCompletion:
```

Same for others